### PR TITLE
fixed regex for expected error messages

### PR DESCRIPTION
**Pull request by yitam**
for calling sqlsrv_fetch right after sqlsrv_cancel

### CODE REVIEW

File: test/sqlsrv/sqlsrv_statement_cancel.phpt [hunk 1]
```diff
@@ -94,7 +94,7 @@ Repro();
 --EXPECTREGEX--
 ﻿
 Test begins...
-\[Microsoft\](\[ODBC Driver 13 for SQL Server\]|\[ODBC Driver Manager\])([ ]{0,1}Function sequence error)
+(.*\[Driver Manager\]|\[Microsoft\](\[ODBC Driver 13 for SQL Server\]|\[ODBC Driver Manager\]))([ ]{0,1}Function sequence error)
```
Comment (ulvii) [inline test/sqlsrv/sqlsrv_statement_cancel.phpt]: Why do we need to do this?

File: test/bvt/sqlsrv/msdn_sqlsrv_cancel.phpt [hunk 2]
```diff
@@ -46,4 +46,4 @@ echo "$count sales accounted for the first $$salesTotal in revenue.\n";
 sqlsrv_cancel( $stmt);
 ?>
 --EXPECT--
-58 sales accounted for the first $104205.9607 in revenue.
\ No newline at end of file
+57 sales accounted for the first $104171.7607 in revenue.
```
Comment (ulvii) [inline test/bvt/sqlsrv/msdn_sqlsrv_cancel.phpt]: Why?

Comment (yitam) [inline test/bvt/sqlsrv/msdn_sqlsrv_cancel.phpt]: It's the problem with the sql query itself. See #436 for details. Back then I put the incorrect expected output.

File: test/sqlsrv/sqlsrv_statement_cancel.phpt (hunk 1 above)
Comment (yitam) [inline test/sqlsrv/sqlsrv_statement_cancel.phpt]: in Mac the error message is different from the others even though the real message `Function sequence error` is the same in all platforms

Comment (ulvii) [inline test/sqlsrv/sqlsrv_statement_cancel.phpt]: UnixODBC 2.3.4 adds extraneous DM prefix to diagnostic messages.

**Pull request merged**